pointcloud/basic_octree_test.go

```diff
@@ -598,3 +630,55 @@ func testPCDToBasicOctree(t *testing.T, artifactPath string) {
 
 	validateBasicOctree(t, basicOct, basicOct.center, basicOct.sideLength)
 }
+
+func TestCachedMaxProbability(t *testing.T) {
+	center := r3.Vector{X: 0, Y: 0, Z: 0}
+	side := 2.0
+
+	octree, err := createNewOctree(center, side)
+	test.That(t, err, test.ShouldBeNil)
+
+	t.Run("get the max prob from an octree", func(t *testing.T) {
+		pointsAndData := []PointAndData{
+			{P: r3.Vector{X: 0, Y: 0, Z: 0}, D: NewValueData(2)},
+			{P: r3.Vector{X: .5, Y: 0, Z: 0}, D: NewValueData(3)},
+			{P: r3.Vector{X: .5, Y: 0, Z: .5}, D: NewValueData(10)},
+			{P: r3.Vector{X: .5, Y: .5, Z: 0}, D: NewValueData(1)},
+			{P: r3.Vector{X: .55, Y: .55, Z: 0}, D: NewValueData(4)},
+			{P: r3.Vector{X: -.55, Y: -.55, Z: 0}, D: NewValueData(5)},
+			{P: r3.Vector{X: .755, Y: .755, Z: 0}, D: NewValueData(6)},
+		}
+
+		err = addPoints(octree, pointsAndData)
+		test.That(t, err, test.ShouldBeNil)
+
+		validateBasicOctree(t, octree, octree.center, octree.sideLength)
+
+		mp := octree.MaxProb()
+		test.That(t, mp, test.ShouldEqual, 0.1)
+
+		mp = octree.node.children[0].MaxProb()
+		test.That(t, mp, test.ShouldEqual, 0.05)
+	})
+	t.Run("cannot set arbitrary values into the octree", func(t *testing.T) {
```

Comment:
```
d := NewValueData(101)
node := newLeafNodeFilled(r3.Vector{}, d)
```
here we are trying to create a `basicOctreeNode` of type `leafNodeFilled`.
because the passed in data value is greater than 100 the node's max probability is set to `emptyProb` == -1.
It would be erroneous to have max probability of 1.01.
then down below (in the test) we have the expected outcome of `newLeafNodeFilled(r3.Vector{}, d)`
```
		filledNode := basicOctreeNode{
			children: nil,
			nodeType: leafNodeFilled,
			point:    PointAndData{P: r3.Vector{}, D: d},
			maxProb:  -1,
		}
```